Astronomy/Networking/MarsRoverClient.swift

```diff
@@ -43,9 +48,9 @@ class MarsRoverClient {
     // MARK: - Private
     
     private func fetch<T: Codable>(from url: URL,
-                           using session: URLSession = URLSession.shared,
+                           using session: NetworkDataLoader = URLSession.shared,
                            completion: @escaping (T?, Error?) -> Void) {
-        session.dataTask(with: url) { (data, response, error) in
+        networkLoader.loadData(from: url) { data, error in
```

Comment:
We may want to leave the response in for the future, in case we need it. It can always be represented with an `_` in the worst case.